src/replication.c

```diff
@@ -3184,8 +3184,16 @@ void waitCommand(client *c) {
     }
 
     /* Argument parsing. */
-    if (getLongFromObjectOrReply(c,c->argv[1],&numreplicas,NULL) != C_OK)
+    if (!strcasecmp(c->argv[1]->ptr,"majority")) {
+        numreplicas = listLength(server.slaves)/2 + 1;
```

Comment:
I think major should be `(listLength(server.slaves) + 1)/2`.
3 nodes -> major should be 2 -> 1 replica
4 nodes -> major should be 3 -> 2 replicas
5 nodes -> major should be 3 -> 2 replicas

Master should also be included.

Reply:
Not majority nodes instead of majority replicas

Reply:
In mongodb, majority means majority of all nodes include primary.

Reply:
Yes, i know, but in redis, WAIT $replicas_number $timeout. we can discuss again when we have a way to solve major troubles as discussed below.